src/clj_http/client.clj

```diff
@@ -1168,12 +1168,26 @@
      (throw (IllegalArgumentException. "Host URL cannot be nil"))))
 
 (defn- request*
-  [req [respond raise]]
-  (if (opt req :async)
+  [req & [respond raise]]
+  {:pre [(not (and (:async req) (:async-future req)))]}
+  (cond
+    (opt req :async)
     (if (some nil? [respond raise])
       (throw (IllegalArgumentException.
               "If :async? is true, you must pass respond and raise"))
       (request (dissoc req :respond :raise) respond raise))
+
+    (opt req :async-future)
+    (let [basic-future (org.apache.http.concurrent.BasicFuture. nil)]
+      (request (-> req
+                   (dissoc :async-future)
```

Comment:
In order to avoid having multiple options I think this should be:

```suggestion
                   (dissoc :async :async? :async-future :async-future?)
```